core/model/src/main/java/org/eclipse/sw360/antenna/model/artifact/facts/ArtifactCoordinates.java

```diff
@@ -11,10 +11,29 @@
 
 package org.eclipse.sw360.antenna.model.artifact.facts;
 
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
 public abstract class ArtifactCoordinates<T extends ArtifactCoordinates> implements ArtifactIdentifier<T> {
     public abstract String getName();
+
     public abstract String getVersion();
 
+    public abstract String getType();
+
+    public String getPurl() {
+        return String.format("pkg:%s/%s@%s", percentEncoding(getType()), percentEncoding(getName()),
+                percentEncoding(getVersion()));
+    }
+
+    protected String percentEncoding(String input) {
+        try {
+            return URLEncoder.encode(input, "UTF-8");
```

Comment:
Just `encode` is unfortunately not enough, encoding "+" needs to be handled explicitly, see https://github.com/heremaps/oss-review-toolkit/blob/673c9b817cb71fed1bcd313d51d4b7788fbd2e61/utils/src/main/kotlin/Extensions.kt#L260-L263

Reply:
How was this resolved? Did you forget to push?